examples/material_gallery/lib/gallery/demo.dart

```diff
@@ -0,0 +1,252 @@
+// Copyright 2016 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'package:flutter_markdown/flutter_markdown.dart';
+
+import '../gallery/syntax_highlighter.dart';
+
+class SingleComponentDemoData {
+  SingleComponentDemoData({
+    this.builder,
+    this.exampleCode,
+    this.description,
+    this.onPressedDemo
+  });
+
+  final WidgetBuilder builder;
+  final String exampleCode;
+  final String description;
+  final VoidCallback onPressedDemo;
+}
+
+class ComponentDemoTabData extends SingleComponentDemoData {
+  ComponentDemoTabData({
+    WidgetBuilder builder,
+    String exampleCode,
+    String description,
+    VoidCallback onPressedDemo,
+    this.tabName
+  }) : super(
+    builder: builder,
+    exampleCode: exampleCode,
+    description: description,
+    onPressedDemo: onPressedDemo
+  );
+
+  final String tabName;
+
+  static Map<ComponentDemoTabData, TabLabel> buildTabLabels(List<ComponentDemoTabData> demos) {
+    return new Map<ComponentDemoTabData, TabLabel>.fromIterable(
+      demos,
+      value: (ComponentDemoTabData demo) => new TabLabel(text: demo.tabName)
+    );
+  }
+
+  @override
+  bool operator==(Object other) {
+    if (other.runtimeType != runtimeType)
+      return false;
+    ComponentDemoTabData typedOther = other;
+    return typedOther.tabName == tabName && typedOther.description == description;
+  }
+
+  @override
+  int get hashCode => hashValues(tabName.hashCode, description.hashCode);
+}
+
+class TabbedComponentDemoScaffold extends StatelessWidget {
+  TabbedComponentDemoScaffold({
+    this.title,
+    this.demos
+  });
+
+  final List<ComponentDemoTabData> demos;
+  final String title;
+
+  @override
+  Widget build(BuildContext context) {
+    return new TabBarSelection<ComponentDemoTabData>(
+      values: demos,
+      child: new Scaffold(
+        appBar: new AppBar(
+          title: new Text(title),
+          tabBar: new TabBar<ComponentDemoTabData>(
+            isScrollable: true,
+            labels: ComponentDemoTabData.buildTabLabels(demos)
+          )
+        ),
+        body: new TabbedComponentDemo(demos)
+      )
+    );
+  }
+}
+
+class TabbedComponentDemo extends StatelessWidget {
+  TabbedComponentDemo(this.demos);
+
+  final List<ComponentDemoTabData> demos;
+
+  @override
+  Widget build(BuildContext context) {
+    return new TabBarView<ComponentDemoTabData>(
+      children: demos.map(buildTabView).toList()
+    );
+  }
+
+  Widget buildTabView(ComponentDemoTabData demo) {
+    return new Builder(
```

Comment:
I used that code from the previous example, I assumed it prevented all tabs from building when they weren't in use. But if that's not the case I'll remove it!